pilot/pkg/controllers/ipallocate/ipallocate.go

```diff
@@ -0,0 +1,407 @@
+	// if we wish to replicate v1 assignment behavior for existing serviceentry it should go here and must
+	// include a change to respect the used IPs from the warmed cache begin reconcile
+	c.queue.Run(stop)
+	c.serviceEntryClient.ShutdownHandlers()
+}
+
+// populateControllerDatastructures reads all serviceentries and records any addresses in our ranges as in use
+func (c *IPAllocate) populateControllerDatastructures() {
+	serviceentries := c.serviceEntryClient.List(metav1.NamespaceAll, klabels.Everything())
+	count := 0
+	for _, serviceentry := range serviceentries {
+		count++
+		owner := config.NamespacedName(serviceentry)
+		log.Debugf("%s/%s found during warming", serviceentry.Namespace, serviceentry.Name)
+
+		for _, addr := range serviceentry.Spec.Addresses {
+			a, err := netip.ParseAddr(addr)
+			if err != nil {
+				log.Debugf("unable to parse address %s for %s/%s, received error: %s", addr, serviceentry.Namespace, serviceentry.Name, err.Error())
+				continue
+			}
+			// these are not assigned by us but could conflict with our IP ranges and cause issues
+			if !c.inOurRange(a) {
+				// don't need to worry about this one
+				continue
+			}
+			c.markUsedOrQueueConflict(a, owner)
+		}
+
+		for _, a := range autoallocate.GetV2AddressesFromServiceEntry(serviceentry) {
+			c.markUsedOrQueueConflict(a, owner)
+		}
+	}
+
+	log.Debugf("discovered %v during warming", count)
+}
+
+func (c *IPAllocate) reconcile(a any) error {
+	if nn, ok := a.(types.NamespacedName); ok {
+		return c.reconcileServiceEntry(nn)
+	}
+	if conflict, ok := a.(conflictDetectedEvent); ok {
+		return c.resolveConflict(conflict)
+	}
+
+	return nil
+}
+
+func (c *IPAllocate) reconcileServiceEntry(se types.NamespacedName) error {
+	log.Debugf("reconciling ServiceEntry %s/%s", se.Namespace, se.Name)
+	serviceentry := c.serviceEntryClient.Get(se.Name, se.Namespace)
+	if serviceentry == nil {
+		// probably a delete so we should remove ips from our addresses most likely
+		// TODO: we never actually remove IP right now, likely this should be done a little more slowly anyway to prevent reuse if we are too fast
+		return nil
+	}
+
+	// TODO: also what do we do if we already allocated IPs in a previous reconcile but are not longer meant to
+	// or IP would not be usabled due to an update? write a condition "IP unsabled due to wildcard host or something similar"
+	if !autoallocate.ShouldV2AutoAllocateIP(serviceentry) {
+		// TODO: DRY with warm...
+		// we may have an address in our range so we should check and record it
+		for _, addr := range serviceentry.Spec.Addresses {
+			a, err := netip.ParseAddr(addr)
+			if err != nil {
+				log.Debugf("unable to parse address %s for %s/%s, received error: %s", addr, serviceentry.Namespace, serviceentry.Name, err.Error())
+				continue
+			}
+			// these are not assigned by us but could conflict with our IP ranges and cause issues
+			if !c.inOurRange(a) {
+				// don't need to worry about this one
+				continue
+			}
```

Comment:
we can ignore this check, only keep checing address exist for simplicity .

IN the admission, we can add validating address, should prevent using ips in range of allocator

Reply:
Good idea. TY!